### ISSUE TITLE
AccessibilityNodeInfo setter (isHeading) not mirrored in shadowed tree

### ISSUE THREAD

**Issue by mihaipopa12** [robolectric/robolectric#7327]
It seems that changing certain properties of `AccessibilityNodeInfo` does not reflect correctly in Roboelectric:
```
  @Test
  fun test() {
    val root = AccessibilityNodeInfo.obtain()
    val node = AccessibilityNodeInfo.obtain()
    shadowOf(root).addChild(node)
    node.isHeading = true
    println(node.isHeading) // true
    println(root.getChild(0).isHeading) // false
  }
```
Note how the second print is `false`, while it should be `true` as well.
If we replace `isHeading` by `isScrollable`, `isCheckable`, etc. this will work as expected (both prints will be `true`).

Version:
```
last_upgrade_date {
    year: 2022
    month: 5
    day: 6
}
```
`Android SDK 31`

**Comment by hoisie** [comment]
Just a couple of thoughts on this.

The issue is that `isHeading` is not implemented in [ShadowAccessibilityNodeInfo](https://github.com/robolectric/robolectric/blob/master/shadows/framework/src/main/java/org/robolectric/shadows/ShadowAccessibilityNodeInfo.java). However, after taking a cursory glance at ShadowAccessibilityNodeInfo, that shadow violates the number one rule of shadows, which is you should not shadow pure Java code. When you do that it turns perfectly valid Android code into a fake. I don't know the history behind ShadowAccessibilityNodeInfo, there may be a reason for this, but I suspect there is not.

It would be an easy fix to just implement `isHeading` in `ShadowAccessibilityNodeInfo`.

**Comment by hoisie** [comment]
FYI I have a pending fix for this which deletes some parts of ShadowAccessibilityNodeInfo
